test/run_pass/test_net_httpclient_parse_error.js

```diff
@@ -0,0 +1,53 @@
+/* Copyright 2017-present Samsung Electronics Co., Ltd. and other contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+var assert = require('assert');
+var http = require('http');
+var net = require('net');
+
+var connects = 0;
+var errors = 0;
+
+// Create a TCP server
+var server = net.createServer(function(socket) {
+  if (++connects === 1) {
+    socket.end('HTTP/1.1 302 Object Moved\r\nContent-Length: 0\r\n\r\n');
+  } else {
+    socket.end('Bad http\r\n');
+    server.close();
+  }
+});
+
+server.listen(3085, '127.0.0.1', function() {
+  for (var i = 0; i < 2; i++) {
+    var request = http.request({
+      host: '127.0.0.1',
+      port: 3085,
+      method: 'GET',
+      path: '/'
+    });
+    // The first request will get a parse error while trying to parse the data,
+    // therefor the client will destroy the socket. After the socket is
+    // destroyed the second request will get a socket hang up error.
+    request.on('error', function(err) {
```

Comment:
In my perspective, the first request will NOT get a parse error since the server sends a response with the proper header. It seems successfully ended. So I expect socket hang up error should not occur at the second response. Please clarify my query.